charts/agent-stack-k8s/templates/_helper.tpl

```diff
@@ -0,0 +1,6 @@
+{{/* Generate basic labels */}}
+{{- define "mychart.labels" }}
+  labels:
+    app: {{ .Release.Name }}
+    {{- toYaml $.Values.labels | nindent 8 }}
```

Comment:
It's still going to be the case that if the user specifies, e.g., `--set labels.app=orcas` that the labels don't match the selectors. This is the error you get when you try to upgrade an existing release
```
Error: UPGRADE FAILED: cannot patch "agent-stack-k8s" with kind Deployment: Deployment.apps "agent-stack-k8s" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"app":"orcas"}: `selector` does not match template `labels`
```
And this is what you get when you try to install a new release:
```
Release "agent-stack-k8s" does not exist. Installing it now.
Error: 1 error occurred:
	* Deployment.apps "agent-stack-k8s" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"app":"orcas"}: `selector` does not match template `labels`
```

There are few ways to approach this.

You can simply reverse the order of the `app: ...` and `{{- toYaml ...` lines. This will give the later one preference. You don't need a separate `_heplers.tpl` file to do this. But this approach also has a footgun as it silently overwrites the `app` value that the user supplied. Furthermore, it leads to the output of `helm template` containing duplicate keys, which is [technically invalid yaml](https://yaml.org/spec/1.2.2/#mapping). Some users might process the output of `helm template` with yaml parsers that would reject it for this reason, so I don't think this is acceptable for us in the long term.

A better approach is to use helper functions to detect when the user is specifying an `app` label and return an error to the user letting them know that `app` is reserved. That's where helper functions come in. You should be able to use something like this: https://masterminds.github.io/sprig/flow_control.html to achieve this.

Reply:
Also, typically, you want there to be no indentation in the helper function so that all the indentation is managed in the template with `nIndent`.